test/e2e/util/kubectl.go

```diff
@@ -104,6 +116,43 @@ func KubectlDescribe(kubeconfigPath, podName, namespace string) (string, error)
 	return Kubectl(args)
 }
 
+// KubectlDescribe executes "kubectl describe" given a list of arguments.
+func KubectlCurlPod(kubeconfigPath string) (string, error) {
+	args := []string{
+		"run",
+		"temp",
+		"--image",
+		"curlimages/curl",
+		"--",
+		"tail",
+		"-f",
+		"/dev/null",
+	}
+	return Kubectl(args)
+}
+
+func KubectlDescribeService(kubeconfigPath, serviceName, namespace string) (string, error) {
```

Comment:
You don't need this, as long as the curl pod is in the same namespace as the otel collector, you can use `http://otel-collector:8889/metrics` as the endpoint.